monkey/monkey_island/cc/services/authentication_service.py

```diff
@@ -40,9 +44,16 @@ def register_new_user(self, username: str, password: str):
 
         self._reset_repository_encryptor(username, password)
 
-    def authenticate(self, username: str, password: str):
+    def authenticate(self, username: str, password: str) -> User:
+        registered_user = User.objects.filter(username=username).first()
+
+        if registered_user is None or not registered_user.verify_and_update_password(password):
+            raise IncorrectCredentialsError()
+
         self._unlock_repository_encryptor(username, password)
 
+        return registered_user
```

Comment:
No point in making these changes, as this PR is about registration, not login

Reply:
We made it about both.... 😢

Reply:
If we do login here, then it should use flask-security-too. But since this is not doing that, might as well just delete the changes and do the login on a separate PR